src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1192 @@
+                        return fn.parent;
+                    }
+                    if (isFunctionDeclaration(fn)) {
+                        return fn;
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+
+    function fixIsolatedDeclarationError(node: Node): DiagnosticOrDiagnosticAndArguments | undefined {
+        // Different --isolatedDeclarion errors might result in annotating type on the same node
+        // avoid creating a duplicated fix in those cases
+        if (fixedNodes?.has(node)) return undefined;
+        fixedNodes?.add(node);
+
+        switch (node.kind) {
+            case SyntaxKind.Parameter:
+            case SyntaxKind.PropertyDeclaration:
+            case SyntaxKind.VariableDeclaration:
+                return addTypeToVariableLike(node as ParameterDeclaration | PropertyDeclaration | VariableDeclaration);
+            case SyntaxKind.ArrowFunction:
+            case SyntaxKind.FunctionExpression:
+            case SyntaxKind.FunctionDeclaration:
+            case SyntaxKind.MethodDeclaration:
+            case SyntaxKind.GetAccessor:
+                return addTypeToSignatureDeclaration(node as SignatureDeclaration, sourceFile);
+            case SyntaxKind.ExportAssignment:
+                return transformExportAssignment(node as ExportAssignment);
+            case SyntaxKind.ClassDeclaration:
+                return transformExtendsClauseWithExpression(node as ClassDeclaration);
+            case SyntaxKind.ObjectBindingPattern:
+            case SyntaxKind.ArrayBindingPattern:
+                return transformDestructuringPatterns(node as BindingPattern);
+            default:
+                throw new Error(`Cannot find a fix for the given node ${node.kind}`);
+        }
+    }
+
+    function addTypeToSignatureDeclaration(func: SignatureDeclaration, sourceFile: SourceFile): DiagnosticOrDiagnosticAndArguments | undefined {
+        if (func.type) {
+            return;
+        }
+        const { typeNode } = inferType(func);
+        if (typeNode) {
+            changeTracker.tryInsertTypeAnnotation(
+                sourceFile,
+                func,
+                typeNode,
+            );
+            return [Diagnostics.Add_return_type_0, typeToStringForDiag(typeNode)];
+        }
+    }
+
+    function transformExportAssignment(defaultExport: ExportAssignment): DiagnosticOrDiagnosticAndArguments | undefined {
+        if (defaultExport.isExportEquals) {
+            return;
+        }
+
+        const { typeNode } = inferType(defaultExport.expression);
+        if (!typeNode) return undefined;
+        changeTracker.replaceNodeWithNodes(sourceFile, defaultExport, [
+            factory.createVariableStatement(
+                /*modifiers*/ undefined,
+                factory.createVariableDeclarationList(
+                    [factory.createVariableDeclaration(
+                        "__default",

```

Comment:
Declaration emit uses a single underscore when we generate this (not amazing) name. Does it need `factory.createUniqueName` as well?

```suggestion
                        factory.createUniqueName("_default"),
```

Reply:
Done